BedrockServer.cpp

```diff
@@ -1930,13 +1938,16 @@ void BedrockServer::_postPollPlugins(fd_map& fdm, uint64_t nextActivity) {
     // Only pass timeouts for transactions belonging to timed out commands.
     uint64_t now = STimeNow();
     map<SHTTPSManager::Transaction*, uint64_t> transactionTimeouts;
-    auto timeoutIt = _outstandingHTTPSCommands.begin();
-    while (timeoutIt != _outstandingHTTPSCommands.end() && (*timeoutIt)->timeout() < now) {
-        // Add all the transactions for this command, even if some are already complete, they'll just get ignored.
-        for (auto transaction : (*timeoutIt)->httpsRequests) {
-            transactionTimeouts[transaction] = (*timeoutIt)->timeout();
+    {
+        lock_guard<mutex> lock(_httpsCommandMutex);
```

Comment:
i assume we weren't locking here before because we only call this function from the sync thread? But now we're locking because we could check the values in canStandDown?

Reply:
I think we should have done this before, too.